src/Microsoft.Identity.Web/TokenAcquisition.Logger.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// LoggingMessage class for TokenAcquisition.
+    /// </summary>
+    internal partial class TokenAcquisition
+    {
+        private static class Logger
+        {
+            private static readonly Action<ILogger, string, Exception> s_tokenAcquisitionError =
+           LoggerMessage.Define<string>(
+               LogLevel.Information,
+               new EventId(1, "TokenAcquisitionError"),
+               "[MsIdWeb] An error occured during token acquisition: {MsalErrorMessage}");
+
+            /// <summary>
+            /// Logger for handling MSAL exceptions in TokenAcquisition.
+            /// </summary>
+            /// <param name="logger">ILogger.</param>
+            /// <param name="msalErrorMessage">Specific log message from TokenAcquisition.</param>
+            /// <param name="ex">Exception from MSAL.NET.</param>
+            public static void TokenAcquisitionError(
+                ILogger logger,
+                string msalErrorMessage,
+                Exception ex) => s_tokenAcquisitionError(logger, msalErrorMessage, ex);
```

Comment:
Why do you need an action here? you can just call the `LoggerMessage.Define<string>..` directly

Reply:
Ok, reading the docs you pointed to :)

Reply:
following the asp.net core [guidelines](https://github.com/dotnet/aspnetcore/wiki/Engineering-guidelines#logging-patterns), so the log template is compiled once and then we use the delegate to invoke it, when needed.